### PR TITLE
Add documentation on Snaps Custom UI (#617)

### LINKED ISSUE #617: Document the "Custom UI" concept

### PR DESCRIPTION
**Pull request by ziad-saab**
This PR adds documentation explaining what the Snaps Custom UI concept is, and how and where to use it. Fixes #617

### CODE REVIEW

File: docs/guide/snaps.md [hunk 1]
```diff
@@ -63,9 +63,7 @@ Bring your insights, anti-phishing, and security solutions to the MetaMask UI wi
 
 #### Custom UI in MetaMask using a defined set of components
 
-Display custom UI within MetaMask using a set of pre-defined components, including Markdown, form controls, and images. This custom UI can include actionable controls for dynamic interfaces that respond to user input.
-
-<img src="../assets/soon.png" alt="Coming soon" style="width: 106px; position: relative; top: 20px;" />
```
Comment (Montoya) [inline docs/guide/snaps.md:68]: Wrong one. This coming soon is for Cron Jobs below it.

File: docs/guide/snaps.md [hunk 2]
```diff
@@ -63,9 +63,7 @@ Bring your insights, anti-phishing, and security solutions to the MetaMask UI wi
 
 #### Custom UI in MetaMask using a defined set of components
```
Comment (Montoya) [inline docs/guide/snaps.md:64]: Above this should be the image for "Live in MetaMask Flask"

File: docs/guide/snaps.md [hunk 3]
```diff
@@ -63,9 +63,7 @@ Bring your insights, anti-phishing, and security solutions to the MetaMask UI wi
...
 
-Display custom UI within MetaMask using a set of pre-defined components, including Markdown, form controls, and images. This custom UI can include actionable controls for dynamic interfaces that respond to user input.
-
-<img src="../assets/soon.png" alt="Coming soon" style="width: 106px; position: relative; top: 20px;" />
+Display [custom UI](./snaps-concepts.md#custom-ui) within MetaMask using a set of pre-defined components, including Markdown, form controls, and images. This custom UI can include actionable controls for dynamic interfaces that respond to user input.
```
Comment (Montoya) [inline docs/guide/snaps.md:66]: Change "including Markdown, form controls, and images" to "including inline Markdown"

Comment (Montoya) [reply docs/guide/snaps.md:66]: remove the sentence after it, that's not supported yet.

**Comment by ziad-saab** [comment]
Added this alongside the Transaction Insights documentation adjustments.

**Pull request closed without merging**